spec/controllers/api/v0/base_controller_spec.rb

```diff
@@ -0,0 +1,166 @@
+    before do
+      request.headers["Authorization"] = "Bearer #{jwt}"
+    end
+
+    describe "#get_api_client_mef_credentials" do
+      it "gets the credentials corresponding to the client app name in the jwt and converts keys to symbols" do
+        secrets_hash = {"mef_env" => "test", "app_sys_id" => "foo", "etin" => "bar", "cert_base64" => "baz"}
+        allow_any_instance_of(Aws::SecretsManager::Client)
+          .to receive(:get_secret_value)
+          .with(secret_id: "efiler-api-client-mef-credentials/#{client_app_name}")
+          .and_return(Aws::SecretsManager::Types::GetSecretValueResponse.new(secret_string: secrets_hash.to_json))
+
+        expect(subject.get_api_client_mef_credentials)
+          .to eq({mef_env: "test", app_sys_id: "foo", etin: "bar", cert_base64: "baz"})
+      end
+    end
+
+    describe "#verify_client_name_and_signature" do
+      it "verifies the request has a valid JWT with cert and client app name" do
+        allow(subject).to receive(:get_api_client_mef_credentials).and_return({efiler_api_public_key: Base64.encode64(PUBLIC_KEY), app_sys_id: "foo", etin: "bar"})
+
+        expect {
+          subject.verify_client_name_and_signature
+        }.not_to raise_error
+      end
+
+      it "raises JWT::VerificationError when cert is not valid" do
+        allow(subject).to receive(:get_api_client_mef_credentials).and_return({efiler_api_public_key: Base64.encode64(WRONG_PUBLIC_KEY), app_sys_id: "foo", etin: "bar"})
+
+        expect {
+          subject.verify_client_name_and_signature
+        }.to raise_error(JWT::VerificationError)
+      end
+    end
+  end
+
+  context "error handling" do
+    before do
+      allow_any_instance_of(described_class).to receive(:verify_client_name_and_signature).and_return(true)
+    end
+
+    context "when it encounters a retryable error" do
+      controller do
+        def index
+          raise MefService::RetryableError
+          head :ok
+        end
+      end
+
+      it "returns a bad gateway response" do
+        get :index
+
+        expect(response.body).to eq("Error contacting MeF, please try again")
+        expect(response.status).to eq(502)
+      end
+    end
+
+    context "when it encounters a missing parameters error" do
+      controller do
+        def index
+          params = ActionController::Parameters.new(a: {})
+          params.fetch(:boop)
+          head :ok
+        end
+      end
+
+      it "returns a bad request response" do
+        get :index
+
+        expect(response.body).to eq("param is missing or the value is empty or invalid: boop")
+        expect(response).to be_bad_request
+      end
+    end
+
+    xcontext "when it encounters an aws resource not found error" do
+      controller do
+        def index
+          raise Aws::SecretsManager::Errors::ResourceNotFoundException
```

Comment:
this one is weird? the current state causes it to give me this:
```
ArgumentError: wrong number of arguments (given 0, expected 2..3)

  0) FakeController error handling when it encounters an aws resource not found error returns unauthorized
     Failure/Error: raise Aws::SecretsManager::Errors::ResourceNotFoundException
```

and when i `raise Aws::SecretsManager::Errors::ResourceNotFoundException(0, 0)` it gives this instead:
```
NoMethodError: undefined method 'ResourceNotFoundException' for module Aws::SecretsManager::Errors

  0) FakeController error handling when it encounters an aws resource not found error returns unauthorized
     Failure/Error: raise Aws::SecretsManager::Errors::ResourceNotFoundException(nil, nil)
```